server/src/test/kotlin/fi/oph/ludos/certificate/CertificateControllerTest.kt

```diff
@@ -0,0 +1,214 @@
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+import java.sql.Timestamp
+import javax.transaction.Transactional
+
+fun postCertificate(body: String) =
+    MockMvcRequestBuilders.post("${Constants.API_PREFIX}/certificate").contentType(MediaType.APPLICATION_JSON)
+        .content(body)
+
+fun getCertificate(exam: Exam, id: Int) =
+    MockMvcRequestBuilders.get("${Constants.API_PREFIX}/certificate/$exam/$id").contentType(MediaType.APPLICATION_JSON)
+
+fun updateCertificate(id: Int, body: String) =
+    MockMvcRequestBuilders.put("${Constants.API_PREFIX}/certificate/$id").contentType(MediaType.APPLICATION_JSON)
+        .content(body)
+
+@TestPropertySource(locations = ["classpath:application.properties"])
+@SpringBootTest
+@AutoConfigureMockMvc
+@Transactional
+class CertificateControllerTest(@Autowired val mockMvc: MockMvc) {
+    val objectMapper = jacksonObjectMapper()
+
+    data class TestCertificateIn(
+        val exam: Exam,
+        val nameFi: String,
+        val contentFi: String,
+        val publishState: PublishState,
+        val fileName: String,
+        val fileKey: String,
+        val fileUploadDate: String,
+    )
+
+    data class TestCertificateOut(
+        val id: Int,
+        val exam: Exam,
+        val nameFi: String,
+        val contentFi: String,
+        val publishState: PublishState,
+        val fileName: String,
+        val fileKey: String,
+        val fileUploadDate: String,
+        val createdAt: Timestamp,
+        val updatedAt: Timestamp
+    )
+
+    @Test
+    @WithYllapitajaRole
+    fun createCertificateTest() {
+        val testCertificate = TestCertificateIn(
+            exam = Exam.SUKO,
+            nameFi = "Test Certificate FI",
+            contentFi = "Certificate content Fi",
+            publishState = PublishState.PUBLISHED,
+            fileName = "test_certificate.pdf",
+            fileKey = "https://amazon_url.com/test_certificate.pdf",
+            fileUploadDate = "2023-06-13"
+        )
+
+        val testCertificateStr = objectMapper.writeValueAsString(testCertificate)
+
+        val postResult = mockMvc.perform(postCertificate(testCertificateStr)).andExpect(status().isOk)
+            .andReturn().response.contentAsString
+
+        val certificateIn = objectMapper.readValue(postResult, TestCertificateOut::class.java)
+
+        assertEquals(testCertificate.nameFi, certificateIn.nameFi)
+        assertEquals(testCertificate.exam, certificateIn.exam)
+        assertEquals(testCertificate.contentFi, certificateIn.contentFi)
+        assertEquals(testCertificate.publishState, certificateIn.publishState)
+        assertEquals(testCertificate.fileName, certificateIn.fileName)
+        assertEquals(testCertificate.fileKey, certificateIn.fileKey)
+        assertNotNull(testCertificate.fileUploadDate)
```

Comment:
Mehän tiedetään, mikä `fileUploadDate`n pitäis olla, arvo kannattais tsekata tässä